code/modules/mob/living/silicon/robot/inventory.dm

```diff
@@ -52,6 +52,7 @@
 		module_state_3 = null
 		inv3.icon_state = "inv3"
 	updateicon()
+	usr.hud_used.update_robot_modules_display()

```

Comment:
May be a good idea to test without the `usr.`, to avoid anything wonky involving things like EMPs.